[![cloudevel](img/cloudevel.png)](https://cloudevel.com)

# Compute firewalls.

##  Firewalls.

https://cloud.google.com/vpc/docs/firewalls

In [12]:
gcloud compute networks get-effective-firewalls default --format=json

{[?2004l
  "firewalls": [
    {
      "allowed": [
        {
          "IPProtocol": "tcp",
          "ports": [
            "443"
          ]
        }
      ],
      "creationTimestamp": "2021-10-17T12:10:02.269-07:00",
      "description": "",
      "direction": "INGRESS",
      "disabled": false,
      "id": "5924767337755560933",
      "kind": "compute#firewall",
      "logConfig": {
        "enable": false
      },
      "name": "default-allow-https",
      "network": "https://www.googleapis.com/compute/v1/projects/tercera-generacion/global/networks/default",
      "priority": 1000,
      "selfLink": "https://www.googleapis.com/compute/v1/projects/tercera-generacion/global/firewalls/default-allow-https",
      "sourceRanges": [
        "0.0.0.0/0"
      ],
      "targetTags": [
        "https-server"
      ]
    },
    {
      "allowed": [
        {
          "IPProtocol": "tcp",
          "ports": [
            "80"
          ]
        }
      ],
      "creationTimestamp": "2021

: 1

In [1]:
gcloud compute firewall-rules list

NAME                    NETWORK  DIRECTION  PRIORITY  ALLOW                         DENY  DISABLED
default-allow-http      default  INGRESS    1000      tcp:80                              False
default-allow-https     default  INGRESS    1000      tcp:443                             False
default-allow-icmp      default  INGRESS    65534     icmp                                False
default-allow-internal  default  INGRESS    65534     tcp:0-65535,udp:0-65535,icmp        False
default-allow-rdp       default  INGRESS    65534     tcp:3389                            False
default-allow-ssh       default  INGRESS    65534     tcp:22                              False

To show all fields of the firewall, please show in JSON format: --format=json
To show all fields in table format, please see the examples in --help.



: 1

### Reglas.

* ```Direction```: ```ingress```o ```egress```.
* ```Priority```: 0 a 65535
* ```Action```: ```allow```o ```deny```
* ```Target```: instancias.
* ```Source/destination``` 
* ```Protocol``` y ```port```: ```TCP```, ```UDP```o ```ICMP```.
* ```Enforment status```: ```enabled``` o ```disabled```.

2 reglas por defecto con prioridad 65535:
* Permite la salida a 0.0.0.0/0
* Niega tráfico de entrada a 0.0.0.0/0

#### Reglas al crear una VPC.

    Incoming traffic from any VM instance on the same network
    Incoming TCP traffic on port 22, allowing SSH
    Incoming TCP traffic on port 3389, allowing Microsoft Remote Desktop Protocol (RDP)
    Incoming Internet Control Message Protocol (ICMP) from any source on the network

```
gcloud compute firewall-rules
```

--action
--allow
--description
--destination-ranges
--direction
--network
--priority
--source-ranges
--source-service-accounts
--source-tags
--target-service-accounts
--target-tags

In [2]:
gcloud compute firewall-rules describe default-allow-https

allowed:
- IPProtocol: tcp
  ports:
  - '443'
creationTimestamp: '2021-10-17T12:10:02.269-07:00'
description: ''
direction: INGRESS
disabled: false
id: '5924767337755560933'
kind: compute#firewall
logConfig:
  enable: false
name: default-allow-https
network: https://www.googleapis.com/compute/v1/projects/tercera-generacion/global/networks/default
priority: 1000
selfLink: https://www.googleapis.com/compute/v1/projects/tercera-generacion/global/firewalls/default-allow-https
sourceRanges:
- 0.0.0.0/0
targetTags:
- https-server


: 1

In [8]:
gcloud compute firewall-rules create regla-tercera-generacion --network=tercera-generacion  --allow=tcp:20000-25000

Creating firewall...⠹Created [https://www.googleapis.com/compute/v1/projects/tercera-generacion/global/firewalls/regla-tercera-generacion].
Creating firewall...done.                                                      
NAME                      NETWORK             DIRECTION  PRIORITY  ALLOW            DENY  DISABLED
regla-tercera-generacion  tercera-generacion  INGRESS    1000      tcp:20000-25000        False


: 1

In [6]:
gcloud compute firewall-rules create --help

NAME
    gcloud compute firewall-rules create - create a Compute Engine firewall
        rule

SYNOPSIS
    gcloud compute firewall-rules create NAME
        (--action=ACTION | --allow=PROTOCOL[:PORT[-PORT]],[...])
        [--description=DESCRIPTION]
        [--destination-ranges=CIDR_RANGE,[CIDR_RANGE,...]]
        [--direction=DIRECTION] [--disabled] [--[no-]enable-logging]
        [--logging-metadata=LOGGING_METADATA]
        [--network=NETWORK; default="default"] [--priority=PRIORITY]
        [--rules=PROTOCOL[:PORT[-PORT]],[...]]
        [--source-ranges=CIDR_RANGE,[CIDR_RANGE,...]]
        [--source-service-accounts=EMAIL,[EMAIL,...]]
        [--source-tags=TAG,[TAG,...]]
        [--target-service-accounts=EMAIL,[EMAIL,...]]
        [--target-tags=TAG,[TAG,...]] [GCLOUD_WIDE_FLAG ...]

DESCRIPTION
    gcloud compute firewall-rules create is used to create firewall rules to
    allow/deny incoming/outgoing traffic.

EXAMPLES
    To create a firewall rule allowing incoming TCP traf


        If neither --source-ranges nor --source-tags are specified,
        --source-ranges defaults to 0.0.0.0/0, which means that the rule
        applies to all incoming IPv4 connections from inside or outside the
        network. If both --source-ranges and --source-tags are specified, the
        rule matches if either the range of the source matches --source-ranges
        or the tag of the source matches --source-tags.

        Multiple IP address blocks can be specified if they are separated by
        commas.

     --source-service-accounts=EMAIL,[EMAIL,...]
        The email of a service account indicating the set of instances on the
        network which match a traffic source in the firewall rule.

        If a source service account is specified then neither source tags nor
        target tags can also be specified.

     --source-tags=TAG,[TAG,...]
        A list of instance tags indicating the set of instances on the network
        to which the rule applies if all othe

: 1

## Etiquetas de red.

## Monitoreo del *firewall*.

https://cloud.google.com/vpc/docs/firewall-rules-logging

<p style="text-align: center"><a rel="license" href="http://creativecommons.org/licenses/by/4.0/"><img alt="Licencia Creative Commons" style="border-width:0" src="https://i.creativecommons.org/l/by/4.0/80x15.png" /></a><br />Esta obra está bajo una <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Licencia Creative Commons Atribución 4.0 Internacional</a>.</p>
<p style="text-align: center">&copy; José Luis Chiquete Valdivieso. 2021.</p>